### PR TITLE
[4.0] Fix media path to banners folder

### PR DESCRIPTION
**Pull request by anuragteapot**
### Issue
Try to create a new banner. And select image.
Error. Server internal error.

### Summary of Changes
Fix path in URL.

### Testing Instructions
Try to create a new banner. And select image.

### Expected result

screenshot from 2018-07-20 08-24-44

### Actual result
screenshot from 2018-07-20 08-25-18

Error

### Documentation Changes Required
NO

### CODE REVIEW

**Comment by ghost** [comment]
PR isn't shown at Issue Tracker, so no mark as successfully Test can be done.

**Comment by ghost** [comment]
This PR is now shown in Issue Tracker, but as Issue, not as PR.

**Comment by dgrammatiko** [comment]
@laoneo can you get Kasun to review this. Hardcoding the local adapter doesn't look very promising solution...

**Comment by brianteeman** [comment]
@laoneo any update on this?

**Comment by laoneo** [comment]
Ups, that slipped through my inbox. With the new media storage API, we get now the question if hardcoded paths should still be supported. Because somebody can disable the local adapter and only work with media data from a cloud provider, this will likely crash again. I'm in favor to not support predefined paths as we store now the last opened folder in the local storage of the user anyway. Opinions?

**Comment by kasvith** [comment]
Allon is correct in here, btw as we resolves the URL without adapter prefix even disabling an adapter wont affect articles we use.

About this one need to check whats the url being used to serve the banner, will take a look on pr

**Comment by kasvith** [comment]
Ok Provided solution is not acceptable for me, it uses a hardcoded adapter, even though we provide a method to fetch absolute url to a file, we can get the real url without the adapter prefix.

It need to be done with js like @dgramatikko did with articles

File: layouts/joomla/form/field/media.php [hunk 1]
```diff
@@ -100,7 +100,7 @@
 	: ($link ? $link
 		: 'index.php?option=com_media&amp;tmpl=component&amp;asset='
 		. $asset . '&amp;author=' . $authorId)
-	. '&amp;fieldid={field-media-id}&amp;path=' . $folder);
+	. '&amp;fieldid={field-media-id}&amp;path=local-0:/' . $folder);
 ?>
```
Comment (kasvith) [inline layouts/joomla/form/field/media.php]: Cannot use Hardcoded adapters.

File: layouts/joomla/form/field/media.php [hunk 2]
```diff
@@ -100,7 +100,7 @@
 	: ($link ? $link
 		: 'index.php?option=com_media&amp;tmpl=component&amp;asset='
 		. $asset . '&amp;author=' . $authorId)
-	. '&amp;fieldid={field-media-id}&amp;path=' . $folder);
+	. '&amp;fieldid={field-media-id}&amp;path=local-0:/' . $folder);
```
Comment (kasvith) [inline layouts/joomla/form/field/media.php]: Hardcoded adapters is not a solution for this

**Comment by brianteeman** [comment]
> , we can get the real url without the adapter prefix.

As long as it is not an absolute URL ;)

**Comment by anuragteapot** [comment]
@kasvith Thanks for your review :)

**Comment by dgrammatiko** [comment]
FWIW I didn't do any magic tricks on the other fields, the reality is that the other fields **don't** use the folder option in the XML. Quite honestly I don't see how some client-side code can help up pick the right adapter. I think there must be a trait with some logic that handles that (some basic checks I guess), or just add one more option to the XML: adapter...

**Comment by roland-d** [comment]
@Anu1601CS Is this ready to be tested?

**Comment by anuragteapot** [comment]
> @Anu1601CS Is this ready to be tested?

This is not perfect fix for this. But, it works for now.

We can open a new issue regarding not to hard code local adapter.

**Comment by roland-d** [comment]
@wilsonge Are you OK with fixing the hard coded local adapter in a separate PR or should it be done in this PR?

**Comment by Schmidie64** [comment]
I have tested this item :white_check_mark: successfully on 571a2e0bfaa92e1c409750e73c4915c34ccbdde1<br /><br />For me the patch works fine, without any obvious problems.
**System**: 
PHP Build On: Windows NT 10.0 build 18362 (Windows 10) i586
10.1.37-MariaDB
PHP-Version: 7.3.0
Webserver: Apache/2.4.37 (Win32) OpenSSL/1.1.1a PHP/7.3.0
Joomla! Version: 	Joomla! 4.0.0-alpha12-dev Development [ Amani ] 19-August-2019 13:40 GMT
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/76.0.3809.132 Safari/537.36
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21189">issues.joomla.org/tracker/joomla-cms/21189</a>.</sub>

**Comment by Lyro274** [comment]
I have tested this item :white_check_mark: successfully on 571a2e0bfaa92e1c409750e73c4915c34ccbdde1<br /><br />I followed the instructions provided by @Anu1601CS. Everything worked fine.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21189">issues.joomla.org/tracker/joomla-cms/21189</a>.</sub>

**Comment by ghost** [comment]
Status "Ready To Commit".

**Comment by wilsonge** [comment]
@roland-d i'd rather fix it if it's not some insane amount of work

**Comment by roland-d** [comment]
@Anu1601CS Can you please fix the issue in this PR?

**Comment by anuragteapot** [comment]
> @Anu1601CS Can you please fix the issue in this PR?

Ok i will try.

**Comment by KishoriBKarale** [comment]
I have tested this item :white_check_mark: successfully on 571a2e0bfaa92e1c409750e73c4915c34ccbdde1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21189">issues.joomla.org/tracker/joomla-cms/21189</a>.</sub>

**Comment by shraddhaSankpal27** [comment]
I have tested this item :white_check_mark: successfully on 571a2e0bfaa92e1c409750e73c4915c34ccbdde1<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/21189">issues.joomla.org/tracker/joomla-cms/21189</a>.</sub>

**Comment by laoneo** [comment]
Thanks!

**Comment by wilsonge** [comment]
Captured doing a proper fix for the issue https://github.com/joomla/joomla-cms/issues/26750 which doesn't block any potential cloud providers here

**Pull request merged**